pallets/did/src/did_details.rs

```diff
@@ -344,7 +413,6 @@ impl<T: Config> DidDetails<T> {
 		if let Some(delegation_key) = details.new_delegation_key {
 			new_did_details.update_delegation_key(delegation_key, current_block_number)?;
 		}
```

Comment:
Can we maybe update the deposit inside here, once we set all the keys by calling `calculate_deposit`?